readetexts-port.patch

```diff
@@ -155,23 +146,24 @@ index b623d59..5de0af0 100644
 +            # Remove all temporary files
 +            instance_path = os.path.join(self.get_activity_root(), 'instance')
 +            shutil.rmtree(instance_path, ignore_errors=True, onerror=None)
++ 
  
-         self.metadata['activity'] = self.get_bundle_id()
-         self.metadata['mime_type'] = 'application/zip'
-@@ -1146,7 +1130,8 @@ class ReadEtextsActivity(activity.Activity):
-             self.alert(_('Error'), _('You must enter at least one search word.'))
+     def can_close(self):
+         self.close_requested = True
+@@ -1186,7 +1164,8 @@ class ReadEtextsActivity(activity.Activity):
+                 'You must enter at least one search word.'))
              self.books_toolbar.search_entry.grab_focus()
              return
--        f = open('bookcatalog.txt', 'r')
+-        f = open('bookcatalog.txt', 'rb')
 +        script_dir = os.path.dirname(os.path.realpath(__file__))
```

Comment:
This is not necessary anymore, either. Always try to minimize the changes, so these patches are the smallest possible.